src/Coin.cpp

```diff
@@ -0,0 +1,264 @@
+    switch(coin) {
+    case TWCoinTypeBinance:
+    case TWCoinTypeBitcoinCash:
+    case TWCoinTypeCallisto:
+    case TWCoinTypeDash:
+    case TWCoinTypeEOS:
+    case TWCoinTypeEthereum:
+    case TWCoinTypeEthereumClassic:
+    case TWCoinTypeGo:
+    case TWCoinTypeICON:
+    case TWCoinTypeNimiq:
+    case TWCoinTypePoa:
+    case TWCoinTypeRipple:
+    case TWCoinTypeTezos:
+    case TWCoinTypeThunderToken:
+    case TWCoinTypeTomoChain:
+    case TWCoinTypeTron:
+    case TWCoinTypeVeChain:
+    case TWCoinTypeWanChain:
+    case TWCoinTypeZcash:
+    case TWCoinTypeZcoin:
+        return TWPurposeBIP44;
+    case TWCoinTypeBitcoin:
+    case TWCoinTypeLitecoin:
+        return TWPurposeBIP84;
+    }
+}
+
+TWCurve TW::curve(TWCoinType coin) {
+    switch(coin) {
+    case TWCoinTypeBinance:
+    case TWCoinTypeBitcoin:
+    case TWCoinTypeBitcoinCash:
+    case TWCoinTypeCallisto:
+    case TWCoinTypeDash:
+    case TWCoinTypeEOS:
+    case TWCoinTypeEthereum:
+    case TWCoinTypeEthereumClassic:
+    case TWCoinTypeGo:
+    case TWCoinTypeICON:
+    case TWCoinTypeLitecoin:
+    case TWCoinTypePoa:
+    case TWCoinTypeRipple:
+    case TWCoinTypeTezos:
+    case TWCoinTypeThunderToken:
+    case TWCoinTypeTomoChain:
+    case TWCoinTypeTron:
+    case TWCoinTypeVeChain:
+    case TWCoinTypeWanChain:
+    case TWCoinTypeZcash:
+    case TWCoinTypeZcoin:
+        return TWCurveSECP256k1;
+    case TWCoinTypeNimiq:
+        return TWCurveEd25519;
+    }
+}
+
+DerivationPath TW::derivationPath(TWCoinType coin) {
+    switch (coin) {
+    case TWCoinTypeBinance:
+    case TWCoinTypeBitcoin:
+    case TWCoinTypeBitcoinCash:
+    case TWCoinTypeCallisto:
+    case TWCoinTypeDash:
+    case TWCoinTypeEOS:
+    case TWCoinTypeEthereum:
+    case TWCoinTypeEthereumClassic:
+    case TWCoinTypeGo:
+    case TWCoinTypeICON:
+    case TWCoinTypeLitecoin:
+    case TWCoinTypeNimiq:
+    case TWCoinTypePoa:
+    case TWCoinTypeRipple:
+    case TWCoinTypeTezos:
+    case TWCoinTypeThunderToken:
+    case TWCoinTypeTomoChain:
+    case TWCoinTypeTron:
+    case TWCoinTypeVeChain:
+    case TWCoinTypeWanChain:
+    case TWCoinTypeZcash:
+    case TWCoinTypeZcoin:
+        return DerivationPath(purpose(coin), coin, 0, 0, 0);
+    }
+}
+
+std::string TW::deriveAddress(TWCoinType coin, const PrivateKey& privateKey) {
+    switch (coin) {
+    case TWCoinTypeBinance:
+        return Tendermint::Address(HRP_BINANCE, privateKey.getPublicKey(PublicKeyType::secp256k1)).string();
+
+    case TWCoinTypeBitcoin:
+        return Bitcoin::Bech32Address(privateKey.getPublicKey(PublicKeyType::secp256k1), 0, HRP_BITCOIN).string();
+
+    case TWCoinTypeBitcoinCash:
+        return Bitcoin::CashAddress(privateKey.getPublicKey(PublicKeyType::secp256k1)).string();
+
+    case TWCoinTypeDash:
+        return Bitcoin::Address(privateKey.getPublicKey(PublicKeyType::secp256k1), TWP2PKHPrefixDash).string();
+
+    case TWCoinTypeCallisto:
+    case TWCoinTypeEOS:
```

Comment:
EOS should not Ethereum::Address

Reply:
Will remove this case since we don't support yet.